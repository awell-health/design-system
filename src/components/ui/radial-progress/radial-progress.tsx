import * as React from 'react';

import { cn } from '@/lib/utils';

interface Props {
  className?: string;
  children?: React.ReactNode | string | JSX.Element;
  value: number;
  style?: object;
}

const RadialProgress: React.FC<Props> = ({ value, className, children, style, ...props }) => {
  const styles = { '--value': value, ...style } as React.CSSProperties;

  return (
    <div className={cn('radial-progress', className)} style={styles} {...props}>
      {children}
    </div>
  );
};

RadialProgress.displayName = 'RadialProgress';

export { RadialProgress };
