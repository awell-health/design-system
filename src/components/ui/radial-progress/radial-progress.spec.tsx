import React from 'react';
import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import { RadialProgress } from './radial-progress';

describe('RadialProgres', () => {
  it('match snapshot', () => {
    const { container } = render(
      <RadialProgress value={30}>
        <div>RadialProgress Content</div>
      </RadialProgress>
    );

    expect(container).toMatchSnapshot();
  });
});
