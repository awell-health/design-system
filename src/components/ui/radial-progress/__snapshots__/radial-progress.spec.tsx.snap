// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RadialProgres > match snapshot 1`] = `
<div>
  <div
    class="radial-progress"
    style="--value: 30;"
  >
    <div>
      RadialProgress Content
    </div>
  </div>
</div>
`;
